### PR TITLE
Create network stats controller for monitor plugin - Closes #5885

### LINKED ISSUE #5885: Create GET api/stats/network for Peer connection statistics
### Description

Main idea is to show connectivity of a node that includes incoming and outgoing connections. This covers, peer connection stats, peer banning and connect/disconnect.

- Expose an endpoint from the p2p library to show a count of incoming/outgoing transactions. Modify existing `getConnectedPeers` to add "connectionKind" property.
- Count the number `EVENT_BAN_PEER` event and capture the `peerId` (peerId is `ipAddress:port`) and the time of event. Add `timeUntilUnBan: (TimeReceived + 24 hours - Time Now)` to know when it will get unbanned and also add a timeout of 24 hours for each entry to delete it when unbanned.
- In the plugin, a function to calculate majority height and count, counter for connect/disconnect peer events

**Response:**

```typescript
{
	Outgoing: {
		count: number,
		networkHeight: {
		  majorityHeight: <height>,
		  numberOfPeers: <count of peers with majority height >
                },
		connectStats: {
		  connects: <count the number of peer connect events >,
		  disconnects: <count the number of peer disconnect events >
           }
	},
	incoming: {
		count: number,
		networkHeight: {
		  majorityHeight: <height>,
		  numberOfPeers: <count of peers with majority height >
                },
		connectStats: {
		  connects: <count the number of peer connect events >,
		  disconnects: <count the number of peer disconnect events >
		},
	},
	totalPeers: {
		connected: getConnectedPeers.count,
		disconnected: getDisconnected.count
	},
        banning: {
	  totalBannedPeers: <count>,
	  bannedPeers: {
		<peerID>:
		{
		  timeUntilUnBan: (TimeReceived + 24 hours - Time Now),
		  Reason: reason,
		  banCount: count
		}, ...
	}
    }
}
```

### Motivation

Developer to get a clear picture of incoming/outgoing connections in a node including stats related to banning, connect/disconnect and networkHeight. 

### Acceptance Criteria

- Add unit test to check if the above values are exposed
- Check if the above values are responded by registering to the framework

### Additional Information

Related issues: #5216

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #5885 

### How was it solved?

🌱 Add getNetworkStats to lisk-p2p 982cbf2
🌱 Add network controller for networkStats 6f7957f
♻️ Add count for incoming and outgoing 79c95ba
✅ Add functional and unit tests for network stats 45d9591
🌱 Add majorityHeight calculation and property to networkStats ad69748
✅ Add unit test for networkStats controller ee1db02

### How was it tested?

Run `npm run test:unit`, `npm run test:integration`, `npm run test:functional` under `elements/lisk-p2p`
Run `npm run test:unit` under `framework-plugins/lisk-framework-monitor-plugin/`

### CODE REVIEW

File: elements/lisk-p2p/src/p2p.ts [hunk 1]
```diff
@@ -260,8 +262,36 @@ export class P2P extends EventEmitter {
...
+			totalRequestsReceived: {},
+		};
+
 		// This needs to be an arrow function so that it can be used as a listener.
 		this._handlePeerPoolRPC = (request: P2PRequest): void => {
+			if (!this._networkStats.totalRequestsReceived[request.procedure]) {
+				this._networkStats.totalRequestsReceived[request.procedure] = 0;
+			}
+
+			this._networkStats.totalRequestsReceived[request.procedure] += 1;
```
Comment (nazarhussain) [inline elements/lisk-p2p/src/p2p.ts:294]: A suggestion if you like this style... 

```js
this._networkStats.totalRequestsReceived[request.procedure] = ++this._networkStats.totalRequestsReceived[request.procedure] || 1;
```

Comment (nazarhussain) [inline elements/lisk-p2p/src/p2p.ts:294]: A suggestion if you like this style... 

```js
this._networkStats.totalRequestsReceived[request.procedure] = ++this._networkStats.totalRequestsReceived[request.procedure] || 1;
```

File: framework-plugins/lisk-framework-monitor-plugin/src/controllers/network.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+const getMajorityHeight = (peers: PeerInfo[]): { height: number; count: number } => {
+	const heightHistogram = {} as { [key: number]: number };
+	const majority = {
+		height: 0,
+		count: 0,
+	};
+	for (const { options } of peers) {
+		const height = options.height as number;
+		heightHistogram[height] = heightHistogram[height] ? heightHistogram[height] + 1 : 1;
```
Comment (nazarhussain) [inline framework-plugins/lisk-framework-monitor-plugin/src/controllers/network.ts:26]:
```suggestion
		heightHistogram[height] = ++heightHistogram[height] || 1;
```

**Review by nazarhussain — APPROVED** [review]

File: elements/lisk-p2p/src/p2p.ts (hunk 1 above)
Comment (ishantiw) [reply elements/lisk-p2p/src/p2p.ts]: Changed it to logical op, not using unary though as it throws lint error

**Review by mitsuaki-u — APPROVED** [review]
LGTM. There is one formatting issue.

**Pull request merged**